src/mito2/src/flush.rs

```diff
@@ -487,8 +501,12 @@ impl FlushScheduler {
         let version = version_control.current().version;
         if version.memtables.is_empty() {
             debug_assert!(!self.region_status.contains_key(&region_id));
-            // The region has nothing to flush.
-            task.on_success();
+            // The region has nothing to flush. Submit a job to update high watermark.
+            let job = task.into_update_high_watermark_job();
+            if let Err(e) = self.scheduler.schedule(job) {
+                error!(e; "Update high watermark for region {}", region_id);
+                return Err(e);
+            }
```

Comment:
Do we need to update it again?

Reply:
Next time when we recieve a flush request it will be updated if the memtable is still empty.